runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WindowingStrategyTranslation.java

```diff
@@ -287,7 +288,8 @@ public static FunctionSpec toProto(WindowFn<?, ?> windowFn, SdkComponents compon
     FunctionSpec windowFnSpec = toProto(windowFn, components);
     String environmentId =
         Strings.isNullOrEmpty(windowingStrategy.getEnvironmentId())
-            ? components.getOnlyEnvironmentId()
+            ? components.getEnvironmentIdFor(
+                ResourceHints.create()) // TODO: get this from elsewhere
```

Comment:
OK, this we need to resolve. I'll do it.

Reply:
I don't remember doing something special for windowing strategies in Python, but also not  familiar how environments are used here. Does python need a follow up as well? Is this critical?

Reply:
Given the offline discussion with Cham, it looks like it's fine.

Reply:
Do we still need a TODO?

Reply:
We can drop this TODO.